unstable test in the TestLocalTemporaryTableReplace

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
[2022-07-21T09:22:47.923Z] --- FAIL: TestLocalTemporaryTableReplace (19.27s)

[2022-07-21T09:22:47.923Z] panic: runtime error: invalid memory address or nil pointer dereference [recovered]

[2022-07-21T09:22:47.923Z] 	panic: runtime error: invalid memory address or nil pointer dereference

[2022-07-21T09:22:47.923Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2e57c19]

[2022-07-21T09:22:47.923Z] 

[2022-07-21T09:22:47.923Z] goroutine 350985 [running]:

[2022-07-21T09:22:47.923Z] testing.tRunner.func1.2({0x39d5ea0, 0x5eaf6e0})

[2022-07-21T09:22:47.923Z] 	/usr/local/go/src/testing/testing.go:1389 +0x24e

[2022-07-21T09:22:47.923Z] testing.tRunner.func1()

[2022-07-21T09:22:47.923Z] 	/usr/local/go/src/testing/testing.go:1392 +0x39f

[2022-07-21T09:22:47.923Z] panic({0x39d5ea0, 0x5eaf6e0})

[2022-07-21T09:22:47.923Z] 	/usr/local/go/src/runtime/panic.go:838 +0x207

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/ddl.(*sessionPool).close(0x0)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/ddl/session_pool.go:88 +0x39

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/ddl.(*ddl).close(0xc00224aa00)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/ddl/ddl.go:726 +0x1ae

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/ddl.(*ddl).Stop(0xc00224aa00)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/ddl/ddl.go:582 +0x85

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/domain.(*Domain).Close(0xc00011ef00)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/domain/domain.go:715 +0x71

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/session.(*domainMap).Get.func1()

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/session/tidb.go:88 +0x566

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/util.RunWithRetry(0x1e, 0x1f4, 0xc0005e17f0)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/util/misc.go:69 +0x8a

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/session.(*domainMap).Get(0x6601820, {0x4422fe8, 0xc0020b1220})

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/session/tidb.go:73 +0x268

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/session.createSessionWithOpt({0x4422fe8, 0xc0020b1220}, 0x0)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/session/session.go:2936 +0x59

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/session.createSession(...)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/session/session.go:2932

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/session.runInBootstrapSession({0x4422fe8, 0xc0020b1220}, 0x3f63028)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/session/session.go:2902 +0x45

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/session.BootstrapSession({0x4422fe8, 0xc0020b1220})

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/session/session.go:2794 +0x259

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/tests/realtikvtest.CreateMockStoreAndDomainAndSetup(0xc003d4e1a0, {0x0?, 0x514?, 0x15ad640?})

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/tests/realtikvtest/testkit.go:142 +0x17b

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/tests/realtikvtest.CreateMockStoreAndSetup(...)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/tests/realtikvtest/testkit.go:116

[2022-07-21T09:22:47.923Z] github.com/pingcap/tidb/tests/realtikvtest/sessiontest.TestLocalTemporaryTableReplace(0xc001b84740?)

[2022-07-21T09:22:47.923Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/tests/realtikvtest/sessiontest/session_test.go:668 +0x4f

[2022-07-21T09:22:47.923Z] testing.tRunner(0xc003d4e1a0, 0x3f641e8)

[2022-07-21T09:22:47.923Z] 	/usr/local/go/src/testing/testing.go:1439 +0x102

[2022-07-21T09:22:47.923Z] created by testing.(*T).Run

```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/ddl/ddl.go
+++ b/ddl/ddl.go
@@ -723,7 +723,9 @@ func (d *ddl) close() {
 	if d.delRangeMgr != nil {
 		d.delRangeMgr.clear()
 	}
-	d.sessPool.close()
+	if d.sessPool != nil {
+		d.sessPool.close()
+	}
 	variable.UnregisterStatistics(d)
 
 	logutil.BgLogger().Info("[ddl] DDL closed", zap.String("ID", d.uuid), zap.Duration("take time", time.Since(startTime)))

